jsonschema-core.xml

```diff
@@ -313,43 +313,70 @@
                         </dd>
-                        <dt>annotations</dt>
+                        <dt>Applicator keywords</dt>
                         <dd>
-                            attach information to an instance for application use
+                            A keyword that has one or more subschemas,
+                            that are applied against the instance,
+                            or some value within the instance,
+                            and logically aggregated together.
                         </dd>
-                        <dt>applicators</dt>
+                        <dt>Reserved keywords</dt>
                         <dd>
-                            apply one or more subschemas to a particular location
-                            in the instance, and combine or modify their results
+                            Keywords that are permitted in a schema,
+                            but that do not do anything by themselves,
+                            though their value may be read by or referred to by another keyword.
                         </dd>
-                        <dt>reserved locations</dt>
+                        <dt>Arguments</dt>
                         <dd>
-                            do not directly affect results, but reserve a place
-                            for a specific purpose to ensure interoperability
+                            Arguments are properties in a schema that are used by an adjacent keyword,
+                            and whose meaning may only be defined in the presence of that keyword.
                         </dd>
                     </dl>
                     <t>
-                        Keywords may fall into multiple categories, although applicators
-                        SHOULD only produce assertion results based on their subschemas'
-                        results.  They should not define additional constraints independent
-                        of their subschemas.
+                        Some properties may be classified into multiple categories.
+                        Some keywords are also arguments to other keywords,
+                        and applicator keywords are both assertions and annotations.
                     </t>
                     <t>
-                        Keywords which are properties within the same schema object are referred to as adjacent keywords.
+                        The categories that a keyword belongs to is defined per keyword.
+                        Additionally, there are naming conventions that impose limits or requirements on the categories that keywords belong to:
+                    </t>
+                    <ul>
+                        <li>
+                            A keyword that begins with "annotation-" is never a core keyword or assertion keyword, but may be an annotation or reserved keyword. These keywords MUST NOT affect a validation result.
```

Comment:
I agree with @jdesrosiers that this needs to align with the `x-` decision that we've already made.